aws-provider-controller-manager (v0.2.0) unrecoverable segfault when processing MachineSet w/o `cluster.k8s.io/cluster-name` label.

/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
Built images off of v0.2.0 tag and launched cluster. I created a machineset with the following manifest:
```yaml
apiVersion: "cluster.k8s.io/v1alpha1"
kind: MachineSet
metadata:
  name: workers-medium
spec:
  replicas: 1
  selector:
    matchLabels:
      role: workers
      size: medium
      set: node
  template:
    metadata:
      labels:
        role: workers
        size: medium
        set: node
    spec:
      providerID: aws:////i-074d1e828a3f8abf7
      providerSpec:
        value:
          ami: {}
          apiVersion: awsprovider/v1alpha1
          iamInstanceProfile: nodes.cluster-api-provider-aws.sigs.k8s.io
          instanceType: t2.medium
          keyName: default
          kind: AWSMachineProviderSpec
          kubeadmConfiguration:
            init:
              localAPIEndpoint:
                advertiseAddress: ""
                bindPort: 0
              nodeRegistration: {}
            join:
              caCertPath: ""
              discovery:
                tlsBootstrapToken: ""
              nodeRegistration: {}
          metadata:
            creationTimestamp: null
      versions:
        kubelet: v1.13.3
```
The apiserver did not throw any validation errors and the machineset was created.  The aws-provider-controller-manager falls into a crashloop with the following logs:

```
I0402 01:48:08.481884       1 controller.go:113] Reconciling Machine "workers-medium-d5ck6"
I0402 01:48:08.481907       1 controller.go:238] Machine "workers-medium-d5ck6" in namespace "cluster.k8s.io/cluster-name" doesn't specify "default" label, assuming nil cluster
E0402 01:48:08.482103       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
bazel-out/darwin-fastbuild/bin/external/io_bazel_rules_go/linux_amd64_pure_stripped/stdlib%/src/runtime/asm_amd64.s:522
GOROOT/src/runtime/panic.go:513
GOROOT/src/runtime/panic.go:82
GOROOT/src/runtime/signal_unix.go:390
pkg/cloud/aws/actuators/machine/actuator.go:377
vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go:200
vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
bazel-out/darwin-fastbuild/bin/external/io_bazel_rules_go/linux_amd64_pure_stripped/stdlib%/src/runtime/asm_amd64.s:1333
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12c2092]

goroutine 123 [running]:
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x108
panic(0x15b8600, 0x29428a0)
	GOROOT/src/runtime/panic.go:513 +0x1b9
sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators/machine.(*Actuator).Exists(0xc0005eb900, 0x19d96e0, 0xc0000920c8, 0x0, 0xc0004998c0, 0x0, 0x0, 0x0)
	pkg/cloud/aws/actuators/machine/actuator.go:377 +0xa2
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine.(*ReconcileMachine).Reconcile(0xc0005e3780, 0xc000093710, 0x7, 0xc000043ca0, 0x14, 0x2957180, 0x0, 0x0, 0x0)
	vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go:200 +0x437
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000445ea0, 0x0)
	vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x18f
sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0004a3e90)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0004a3e90, 0x3b9aca00, 0x0, 0x1, 0xc00025e840)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbe
sigs.k8s.io/cluster-api-provider-aws/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0004a3e90, 0x3b9aca00, 0xc00025e840)
	vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/cluster-api-provider-aws/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x32a
```
I'm unable to do anything with the cluster at this point. I'm unable to force delete the machines or bring the aws-provider-controller-manager back into a ready state. At this point the only way forward is to tear the cluster down and start over. 
**What did you expect to happen:**
I expect:
* the machineset to have a "failed" status or be rejected by the apiserver for having a critical missing field OR assign a sane default (e.g. controller adds label `cluster.k8s.io/cluster-name=<current-cluster>`). 
* the aws-provider-controller-manager to handle the bad machineset gracefully and continue functioning. 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Cluster-api-provider-aws version:  v0.2.0
- Kubernetes version: (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.4", GitCommit:"f49fa022dbe63faafd0da106ef7e05a29721d3f1", GitTreeState:"clean", BuildDate:"2018-12-14T07:10:00Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`):

--- a/pkg/cloud/aws/actuators/machine/actuator.go
+++ b/pkg/cloud/aws/actuators/machine/actuator.go
@@ -124,6 +124,9 @@ func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachine
 
 // Create creates a machine and is invoked by the machine controller.
 func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	if cluster == nil {
+		return errors.Errorf("missing cluster for machine %s/%s", machine.Namespace, machine.Name)
+	}
 	klog.Infof("Creating machine %v for cluster %v", machine.Name, cluster.Name)
 
 	scope, err := actuators.NewMachineScope(actuators.MachineScopeParams{Machine: machine, Cluster: cluster, Client: a.client})
@@ -230,6 +233,9 @@ func (a *Actuator) reconcileLBAttachment(scope *actuators.MachineScope, m *clust
 
 // Delete deletes a machine and is invoked by the Machine Controller
 func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	if cluster == nil {
+		return errors.Errorf("missing cluster for machine %s/%s", machine.Namespace, machine.Name)
+	}
 	klog.Infof("Deleting machine %v for cluster %v.", machine.Name, cluster.Name)
 
 	scope, err := actuators.NewMachineScope(actuators.MachineScopeParams{Machine: machine, Cluster: cluster, Client: a.client})
@@ -327,6 +333,10 @@ func (a *Actuator) isMachineOutdated(machineSpec *v1alpha1.AWSMachineProviderSpe
 // If the Update attempts to mutate any immutable state, the method will error
 // and no updates will be performed.
 func (a *Actuator) Update(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	if cluster == nil {
+		return errors.Errorf("missing cluster for machine %s/%s", machine.Namespace, machine.Name)
+	}
+
 	klog.Infof("Updating machine %v for cluster %v.", machine.Name, cluster.Name)
 
 	scope, err := actuators.NewMachineScope(actuators.MachineScopeParams{Machine: machine, Cluster: cluster, Client: a.client})
@@ -374,6 +384,10 @@ func (a *Actuator) Update(ctx context.Context, cluster *clusterv1.Cluster, machi
 
 // Exists test for the existence of a machine and is invoked by the Machine Controller
 func (a *Actuator) Exists(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) (bool, error) {
+	if cluster == nil {
+		return false, errors.Errorf("missing cluster for machine %s/%s", machine.Namespace, machine.Name)
+	}
+
 	klog.Infof("Checking if machine %v for cluster %v exists", machine.Name, cluster.Name)
 
 	scope, err := actuators.NewMachineScope(actuators.MachineScopeParams{Machine: machine, Cluster: cluster, Client: a.client})

